### PR TITLE
[release-1.29] fix: Ensure NSG rules cleanup excludes DestinationApplicationSecurityGroups

### PR DESCRIPTION
**Pull request by zarvd**
#### What type of PR is this?

/kind bug
#### What this PR does / why we need it:
cherry-pick #6331 
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
Ensure NSG rules cleanup excludes DestinationApplicationSecurityGroups
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -14,7 +14,7 @@
 
 # syntax=docker/dockerfile:1
 
-FROM --platform=linux/amd64 golang:1.22-bullseye@sha256:78b171fe51f25b8c3197710f281dacae94759a254e9a486576005f9dadba9e7d AS builder
+FROM --platform=linux/amd64 golang:1.22-bullseye@sha256:0742798b94c656e75ba5409583c2485f18b52de5c9b7cf12859c945bb7414718 AS builder
```
Comment (zarvd) [inline Dockerfile:17]: ref: https://hub.docker.com/layers/library/golang/1.22-bullseye/images/sha256-6653e292cffc69f2a25ab096d70b7871e1db0a7acefae27b9853217e723bdda7?context=explore

**Review by feiskyer — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**